### DISCUSSION TITLE
Getting Started on Mac OS?

### DISCUSSION THREAD

**Discussion by njh** [mfucci/node-matter#277] [General]
Hello,

This seems like a very newbie question but I can't get the basics working!
I wanted to give `node-matter` a go on MacOS so followed the README:
```
$ npm i -g node-matter
```

But when trying to run a CLI server:
```
$ matter
2023-7-4 8:24:2.362 INFO Device node-matter@0.0.10
2023-7-4 8:24:2.372 INFO Device Listening
node:internal/errors:496
    ErrorCaptureStackTrace(err);
    ^

Error: bind EADDRINUSE 127.0.0.1:5353
    at node:dgram:362:20
    at process.processTicksAndRejections (node:internal/process/task_queues:83:21) {
  errno: -48,
  code: 'EADDRINUSE',
  syscall: 'bind',
  address: '127.0.0.1',
  port: 5353
}

Node.js v20.3.0
```

I think this is because on MacOS, `mDNSResponder` is already bound to port 5353 on all interfaces:
```
$ sudo lsof -Pni | grep 5353
mDNSRespo  212 _mdnsresponder    6u  IPv4 0x373fef450a44026b      0t0    UDP *:5353
mDNSRespo  212 _mdnsresponder    7u  IPv6 0x373fef450a440a6b      0t0    UDP *:5353
```

This is probably a similar problem to #184.

But I am not sure what the solution is?
Is anyone else developing on Mac OS?

**Comment by JimBuzbee** [comment]
I use a  Mac with matter.js, but have only done so by cloning and building the repo like so:

$ git clone  https://github.com/project-chip/matter.js.git
$ cd matter.js
$ npm i
$ npm run matter-device

**Reply by njh** [reply]
Thank you very much for your speedy reply.

I managed to get it working easily after following your instructions.
Now I just need to develop my application :)

**Comment by Apollon77** [comment]
I develop it on a Mac and UDP ports are normally never blocked "hard" and my "sudo lsof -Pni | grep 5353" shows even some more that are listening.

But it seems that something is strange with you.

But also honestly I use nodejs 18 and not 20 ... so maybe Nodejs 20 has something special? Can you please try nodejs 18?

> This is probably a similar problem to https://github.com/mfucci/node-matter/issues/184.

No it is not because this is just a strange warning on macs ... just ignore

And lastly: Yes Please switch to matter.js because node-matter is deprecated. see https://github.com/mfucci/node-matter#this-repository-has-been-moved

**Reply by njh** [reply]
Thank you very much.  I managed to get it working fine with the [project-chip/matter.js](https://github.com/project-chip/matter.js) repo.

I have absolutely no idea how I missed the **"THIS REPOSITORY HAS BEEN MOVED!!"** messages!

@mfucci Maybe it would be good to mark this repo as Archived?
